data/data-pipeline/data_pipeline/etl/sources/census_acs/etl_utils.py

```diff
@@ -18,6 +19,113 @@ def _fips_from_censusdata_censusgeo(censusgeo: censusdata.censusgeo) -> str:
     return fips
 
 
+def impute_by_geographic_neighbors(
```

Comment:
This is also not so clear. Open to any tips!

Reply:
This is so confusing. Basically, we want to pull the value for each tract from the value in neighbor tracts.  When there's no values in the neighbor tracts, it pulls the county values. If there are no county values, it pulls the state values. 

It's expensive to compute neighbors -- here, because we are not imputing that many tracts, I don't make a geospatial index or anything. As an aside: When we go to do this for the donut holes, since we will have to find neighbors for ~30k tracts, it will be super expensive and so I had planned to use states first as a simpler version of a geospatial index.

Reply:
Yeah, potentially -- like clusters of neighbors or a geospatial index as a first pass.